lib/livebook/sessions.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Livebook.Sessions do
+  @moduledoc false
+
+  # This module is responsible for starting and discovering sessions.
+  #
+  # Every session has a server process and is described by a `%Session{}`
+  # info struct. Information about all sessions in the cluster is
+  # propagated using `Livebook.Tracker`, which serves as an ephemeral
+  # distributed database for the `%Session{}` structs.
+
+  alias Livebook.{Session, Utils}
+
+  @doc """
+  Spawns a new `Session` process with the given options.
+
+  Makes the session globally visible within the session tracker.
+
+  `{:session_created, session}` gets broadcasted under the "sessions" topic.
+  """
+  @spec create_session(keyword()) :: {:ok, Session.t()} | {:error, any()}
+  def create_session(opts \\ []) do
+    id = Utils.random_node_aware_id()
+
+    opts = Keyword.put(opts, :id, id)
+
+    case DynamicSupervisor.start_child(Livebook.SessionSupervisor, {Session, opts}) do
+      {:ok, pid} ->
+        session = Session.get_by_pid(pid)
+
+        case Livebook.Tracker.track_session(session) do
+          :ok ->
+            {:ok, session}
+
+          {:error, reason} ->
+            Session.close(pid)
+            {:error, reason}
+        end
```

Comment:
I wonder if we should move this to inside the `Session` process. It has the benefit we don't need Session.get_by_pid anymore, less back and forth, and we can return `:ignore` from init if the session already exists.

Reply:
We would still need `get_by_pid` because `Sessions.create_session` return the created session. I thought we could use the `{:ok, state, info}` tuple, but it looks like gen server doesn't support it (?)

Reply:
I see, thanks for explaining it!